tests/cupyx_tests/scipy_tests/spatial_tests/test_distance.py

```diff
@@ -62,17 +67,48 @@ def test_cdist_(self, xp, scp):
             if xp == cupy:
                 a = cupy.asarray(a)
 
+            norm = numpy.sum(b_n, axis=1)
+            b = (b_n.T / norm).T
+            if xp == cupy:
+                b = cupy.asarray(b)
+
         if self.metric == 'minkowski':
-            out = scp.spatial.distance.cdist(a, a, metric=self.metric,
+            out = scp.spatial.distance.cdist(a, b, metric=self.metric,
                                              p=self.p).astype(self.dtype)
         else:
-            out = scp.spatial.distance.cdist(a, a, metric=self.metric)\
+            out = scp.spatial.distance.cdist(a, b, metric=self.metric)\
                 .astype(self.dtype)
 
         print(str(out.shape))
         return out
 
 
+@testing.with_requires("scipy")
+@testing.parameterize(*testing.product({
+    'dtype': ['float32', 'float64'],
+    'rows': [20, 100],
+    'cols': [20, 100],
+    'p': [1.0, 2.0, 3.0],
+    'order': ["C", "F"]
+}))
+@pytest.mark.skipif(cupy.cuda.runtime.is_hip, reason="tests for CUDA only")
+@pytest.mark.skipif(not scipy_available or not pylibraft_available,
+                    reason='requires scipy and pylibraft')
+class TestPdist(unittest.TestCase):
```

Comment:
nit: We use pytest for our unit testing framework, so test classes do not need to inherit unittest.TestCase. (Some tests in CuPy do so just because of historical reasons)

```suggestion
class TestPdist:
```